tests/unit/model-test.js

```diff
@@ -713,6 +713,52 @@ module('unit/model', function(hooks) {
     assert.equal(get(model, 'hb'), true, 'alias to missing with default');
   });
 
+  test('.unknownProperty resolves id-matched values to external m3-models when special ref present in _data passed to schema', function(
+    assert
+  ) {
+    let model = run(() => {
+      return this.store.push({
+        data: {
+          id: 'isbn:9780439708180',
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            name: `Harry Potter and the Sorcerer's Stone`,
+            pubDate: 'September 1989',
+            '*relatedBook': 'isbn:9780439136365',
+          },
+        },
+        included: [
+          {
+            id: 'isbn:9780439136365',
+            type: 'com.example.bookstore.Book',
+            attributes: {
+              name: `Harry Potter and the Order of the Phoenix`,
+            },
+          },
+        ],
+      });
+    });
+
+    SchemaManager.schema.computeAttributeReference = function(key, value, modelName, data) {
+      let refValue = data['*' + key];
+      if (typeof value === 'string' && refValue && refValue === value && /^isbn:/.test(value)) {
```

Comment:
`value` will be `undefined` in this case; we should be ignoring it.

The only reason it's not `undefined` in your test is because of the `set` made on line 753.

I think you meant to `typeof refValue`

Reply:
the conditional here.